__tests__/app.test.js

```diff
@@ -0,0 +1,77 @@
+const request = require("supertest");
+const app = require("../app.js");
+const seed = require('../db/seeds/seed')
+const data = require('../db/data/test-data')
+const db = require('../db/connection.js')
+
+beforeEach(() => seed(data))
+
+afterAll(() => db.end());
+
+describe("GET /*", () => {
+  test("Status 404 when given invalid path", () => {
+      return request(app)
+        .get("/badpath")
+        .expect(404)
+        .then((response) => {
+          expect(response.error.text).toBe("Path not found!")
+        })
+  })
+})
+
+describe("GET /api/topics", () => {
+    test("Status 200", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+    });
+    test("returns an array of objects with properties 'slug', 'description", () => {
+        return request(app)
+            .get("/api/topics")
+            .expect(200)
+        .then((response) => {
+            expect(response.body).toHaveLength(3);
+            response.body.forEach((topic) => {
+                expect(topic).toEqual(
+                  {
+                    description: expect.any(String),
+                    slug: expect.any(String)
+                  })
+              });
+        })
+    })
+});
+
+describe("GET /api/articles/:article_id", () => {
+    test("Status 200", () => {
+      return request(app)
+        .get("/api/articles/3")
+        .expect(200)
+    });
+    test("returns a specified object from the database", () => {
+        return request(app)
+            .get("/api/articles/3")
+            .expect(200)
+        .then((response) => {
+            expect(Object.keys(response.body)).toHaveLength(7);
+            expect(response.body).toEqual({
+                article_id: 3,
+                title: "Eight pug gifs that remind me of mitch",
+                topic: "mitch",
+                author: "icellusedkars",
+                body: "some gifs",
+                created_at: '2020-11-03T09:12:00.000Z',
+                votes: 0,
```

Comment:
we can use expect.objectContaining() to ensure the test doesn't break if more properties are added in the future